### PR TITLE
docs: add patching examples

### PR DESCRIPTION
**Pull request by frezbo**
Add an example to show patching

Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

File: website/content/docs/v0.4/Resource Configuration/metadata.md [hunk 1]
```diff
@@ -13,12 +13,143 @@ which is used during the automated installation.
...
+    - manufacturer: Supermicro
+  selector:
+    matchLabels:
+      metal.sidero.dev/serverclass: general.medium.x86
+```
+
+The base template is constructed from the Talos bootstrap provider, using data from the associated `TalosControlPlane` and `TalosConfigTemplate` manifest.
 Then, any configuration patches are applied from the `ServerClass` and `Server`.
 
 Only configuration patches are allowed in the `ServerClass` and `Server` resources.
```
Comment (rsmitty) [inline website/content/docs/v0.4/Resource Configuration/metadata.md:155]: Can be a future PR, but we should probably remove this line.

Comment (frezbo) [reply website/content/docs/v0.4/Resource Configuration/metadata.md]: fixed

**Review by smira — APPROVED** [review]
:+1:

**Pull request merged**